tests/files/dummy_data.sql

```diff
@@ -45,6 +45,25 @@ INSERT INTO categories (id, category, project_type) VALUES
     (106, 'mobs', 2),
     (107, 'optimization', 2);
 
+-- Create dummy oauth client, secret_hash is SHA512 hash of full lowercase alphabet
+INSERT INTO oauth_clients (
+        id,
+        name,
+        icon_url,
+        max_scopes,
+        secret_hash,
+        created_by
+    )
+VALUES (
+        1,
+        'oauth_client_alpha',
+        NULL,
+        $1,
+        '4dbff86cc2ca1bae1e16468a05cb9881c97f1753bce3619034898faa1aabe429955a1bf8ec483d7421fe3c1646613a59ed5441fb0f321389f77f48a879c7b1f1',
+        3
+    );
+INSERT INTO oauth_client_redirect_uris (id, client_id, uri) VALUES (1, 1, 'https://modrinth.com/oauth_callback');
```

Comment:
why is this included in the migration. it doesn't seem to be needed IMO, since you can create/generate this data via api routes, which should be preferred over raw database insertion

Reply:
Yeah totally agree, I wish I didn't have to. It's because of the client secret and how dummy data works now. We only run the code that interacts with the API to insert dummy data once per dummy data version - so most of the time when you run a test, that code inserting through the API isn't being run. Instead, we're just using methods to get data through the API. Since the client secret is generated by labrinth and isn't stored in its original form for security (we just keep the hash), it's on the creator of the oauth client to remember the secret.

Inserting directly into the database with a hash of a known, hardcoded secret was the best solution I saw to that problem.